package com.malware.library.model;

import com.malware.library.controllers.OrderController;
import com.malware.library.exceptions.UserNotFoundException;
import com.malware.library.repository.BookRepository;
import com.malware.library.repository.OrderBookRepository;
import com.malware.library.repository.OrderRepository;
import com.malware.library.repository.UserRepository;
import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;
import com.malware.library.model.*;
import java.time.LocalDate;
import java.util.*;

@RunWith(SpringRunner.class)
@SpringBootTest
public class OrderControllerUpdateTest {
    @Autowired
    private OrderRepository orderRepository;

    @Autowired
    private OrderBookRepository orderBookRepository;

    @Autowired
    private BookRepository bookRepository;

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private OrderController orderController;

    @Test
    public void updateOrder() throws Exception {
        TreeMap<Long, LocalDate> bookIdDates = new TreeMap<>();
        LocalDate date1 = LocalDate.of(2020, 5,30);
        LocalDate date2 = LocalDate.of(2020, 5,30);
        bookIdDates.put(1L, date1);
        bookIdDates.put(2L, date2);
        orderController.update(5,10,null,null, null, null, null, bookIdDates, 21L);
    }
}
